File: Machine Learning A-Z/Part 2 - Regression/Section 6 - Polynomial Regression/polynomial_regression.R
# Data Preprocessing Template

# Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset=dataset[,2:3]

predData=dataset[1,]
predData[1,1]=6.5

regressor=lm(formula = Salary ~ .,data = dataset)
predData$Salary=predict(object = regressor,newdata = predData)

library(ggplot2)
ggplot()+
  geom_point(aes(dataset$Level,dataset$Salary),color='red')+
  geom_line(aes(dataset$Level,predict(object = regressor,newdata = dataset)),color='blue')+
  ggtitle('Salary Linear Predictor')+
  xlab('Level')+
  ylab('Salary')

dataset$Level1=dataset$Level^2
predData$Level1=predData$Level^2
dataset$Level2=dataset$Level^3
predData$Level2=predData$Level^3
dataset$Level3=dataset$Level^4
predData$Level3=predData$Level^4
regressor=lm(formula = Salary ~ .,data = dataset)
predData$Salary=predict(object = regressor,newdata = predData)

library(ggplot2)
ggplot()+
  geom_point(aes(dataset$Level,dataset$Salary),color='red')+
  geom_line(aes(dataset$Level,predict(object = regressor,newdata = dataset)),color='blue')+
  ggtitle('Salary Polynomial Predictor')+
  xlab('Level')+
  ylab('Salary')

File: Machine Learning A-Z/Part 2 - Regression/Section 6 - Polynomial Regression/polynomial_regression.py
# Data Preprocessing Template

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:,1:2].values
y = dataset.iloc[:, 2].values
level=3

from sklearn.linear_model import LinearRegression
linear=LinearRegression()
regressor=linear.fit(X,y)
x_pred=np.ones((1,1))
x_pred[0,0]=level
linear_pred=regressor.predict(x_pred)

plt.scatter(X,y,color='red')
plt.plot(X,regressor.predict(X),color='blue')
plt.xlabel('Level')
plt.ylabel('Salary')
plt.title('Salary Linear Predictor')
plt.show()

from sklearn.preprocessing import PolynomialFeatures
features=PolynomialFeatures(degree=4)
X_poly=features.fit_transform(X)

linear=LinearRegression()
regressor=linear.fit(X_poly,y)
x_poly_pred=features.fit_transform(x_pred)

poly_pred=regressor.predict(x_poly_pred)
plt.scatter(X,y,color='red')
plt.plot(X,regressor.predict(X_poly),color='blue')
plt.xlabel('Level')
plt.ylabel('Salary')
plt.title('Salary Polynomial Predictor')
plt.show()


print("Linear=",linear_pred," vs Poly=",poly_pred," for a level of ",level)
